### PR TITLE
# 161342844 like or dislike articles

### PR DESCRIPTION
**Pull request by wiztemple**
#### What does this PR do?
- Like and dislike articles
#### Description of tasks to be completed
- build a Reaction model
-  write a method to like an article
- write a method to dislike an article
- write a method to Love and unlove an article
- migrate table
- write test for all edge cases

#### Relevant Pivotal Tracker stories?
- [#161342844](https://www.pivotaltracker.com/story/show/161342844)

#### How can this be manually tested?
- git clone https://github.com/andela/haven-ah-backend.git.
- Install node if not installed.
- ensure that PostgreSQL is running on your local machine.
- run npm install to install the npm packages.
- run npm test to run the unit and integration tests.

#### Screenshots(if appropriate)
screenshot 2018-11-08 at 6 57 56 pm
screenshot 2018-11-08 at 6 57 46 pm
screenshot 2018-11-08 at 6 57 24 pm
screenshot 2018-11-08 at 6 56 53 pm
screenshot 2018-11-08 at 6 56 37 pm

### CODE REVIEW

File: server/controllers/reaction.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * @returns {object} User Object
+   */
+  static async postReaction(request, response) {
+    const { slug } = request.params;
+    const { userId } = request;
+    const { reactionType } = request.body;
+
+    let message;
+    switch (reactionType) {
+      case 'Like':
```
Comment (seyi-adeleke) [inline server/controllers/reaction.js]: Lets have these in a constants file like so
`export const LIKE = 'like' `

File: server/controllers/reaction.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        message = '';
+    }
+    try {
+      const existingArticle = await articleRepo.getArticleBySlug(slug);
+      if (!existingArticle) {
+        return badHttpResponse(response, 404, 'article does not exist');
+      }
+      const existingReaction = await reactionRepo.getReaction(userId, existingArticle.id);
+
+      if (existingReaction) {
```
Comment (seyi-adeleke) [inline server/controllers/reaction.js]: This can be simplified to be `existingReaction && existingReaction.reactionType !== reactionType`

**Review by seyi-adeleke — COMMENTED** [review]
good job so far

File: server/controllers/reaction.js (hunk 2 above)
Comment (wiztemple) [reply server/controllers/reaction.js]: @seyi-adeleke , I'm making two checks there, if the reaction type sent is different from existing reaction, it should update it, but if the reaction type sent is the same as the existing reaction, it should be removed. its a nested IFs.

**Review by TheoOkafor — COMMENTED** [review]
Nice work @wiztemple. 👍

File: server/controllers/reaction.js (hunk 2 above)
Comment (seyi-adeleke) [reply server/controllers/reaction.js]: @wiztemple instead of having nested IF's you can have one if statement with both conditions as I wrote above.

**Review by seyi-adeleke — APPROVED** [review]
